testutil/testdevice_linux.go

```diff
@@ -0,0 +1,97 @@
+// +build linux
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type testDevice struct {
+	opts       TestDeviceOpts
+	fileName   string
+	deviceName string
+}
+
+func NewTestDevice(t *testing.T, opts TestDeviceOpts) TestDevice {
```

Comment:
Shouldnt we test overlay with several fs configurations?
This loopback test device should be useful for that as well